src/components/App/PuzzleGenerate/index.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+import { useEffect, useState } from 'react';
+import styles from './PuzzleGenerate.module.scss';
+import { Button, Input } from '../../Global';
+import { createPuzzleInstance, getAllPuzzles } from '../../../services';
+
+const PuzzleGenerate = () => {
+  const [puzzleList, setPuzzleList] = useState([]);
+  const [hint, setHint] = useState('');
+  const [latitude, setLatitude] = useState('');
+  const [longitude, setLongitude] = useState('');
+  const [address, setAddress] = useState('');
+  const [puzzleId, setPuzzleId] = useState('');
+
+  const handleSubmit = () => {
+    console.log(hint, latitude, longitude, address, puzzleId);
+    createPuzzleInstance(
+      parseInt(puzzleId),
+      parseInt(longitude),
+      parseInt(latitude),
+      address,
+      hint
+    ).then(puzzleInstance => console.log(puzzleInstance));
+  };
+
+  useEffect(() => {
+    const fetchPuzzles = async () => {
+      const puzzles = await getAllPuzzles();
+      setPuzzleList(puzzles);
+      setPuzzleId(puzzles[0].id);
```

Comment:
Why?

Reply:
Also what if the list is empty?

Reply:
The options does not give me the default value, so I do this manually

Reply:
And yes I forgot to handle empty :(